### PR TITLE
Add useESModules option

### PR DESCRIPTION
**Pull request by TrySound**
Ref #168 
Option name chosen in symmetry to [babel-plugin-transform-runtime](https://github.com/babel/babel/tree/master/packages/babel-plugin-transform-runtime#useesmodules).

### CODE REVIEW

**Comment by brigand** [comment]
Thanks for working on this!

I think we should preserve the existing behavior with existing configs, so if useESModules is null/undefined, it'd use the existing deadCode config check. Maybe that's too complex though. I'll go with whatever you think is best here.

After evaluating that, could you also update the README?

**Comment by TrySound** [comment]
@brigand Done. However so much conditions are a bit confusing. We can take care of it later I think.

File: README.md [hunk 1]
```diff
@@ -177,6 +177,10 @@ Example of specifying a custom expression:
   "plugins": [["flow-react-proptypes", { "deadCode": "__PROD__" }]]
 ```
 
+## useESModules
+
+The `useESModules` forces esm with enabled `deadCode` which wraps prop types with condition, but do not remove import/export statement in favor of your favorite bundle.
```
Comment (brigand) [inline README.md]: Maybe

```
The `useESModules` option forces this plugin to output ES6 modules, even if the `deadCode` option is enabled. Your bundler will be responsible for removing the dead code.
```

Comment (TrySound) [reply README.md]: Yep. That's better. I'm lazy for docs.)

**Comment by brigand** [comment]
Looks good, thanks! 15.1.0

**Pull request merged**